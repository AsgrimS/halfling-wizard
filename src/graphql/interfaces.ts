import {
  Class,
  Subclass,
  Proficiency,
  ClassProficiency_Choices,
  AbilityScoreSkills,
  ClassSpellcasting,
  ClassStarting_Equipment,
} from "./schema";

export interface HeroClassesData {
  classes: Class[];
}

export interface HeroClassData {
  class: Class;
}

export interface SubclassesData {
  subclasses: Subclass[];
}

export interface ProficienciesData {
  proficiencies: Proficiency[];
}

export interface ProficienciesChoicesData {
  proficienciesChoices: ClassProficiency_Choices[];
}
export interface SavingThrowsData {
  saving_throws: AbilityScoreSkills[];
}

export interface ClassSpellcastingData {
  spellcasting: ClassSpellcasting;
}

export interface StartingEquipmentData {
  startingEquipments: ClassStarting_Equipment[];
}
