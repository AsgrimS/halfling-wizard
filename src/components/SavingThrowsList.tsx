import React from "react";
import { SavingThrowsData } from "../graphql/interfaces";

const SavingThrowsList: React.FC<SavingThrowsData> = ({ saving_throws }) => {
  return (
    <div>
      <span>Saving Throws:</span>
      <ul>
        {saving_throws.map((saving_throw) => (
          <li key={saving_throw.index}>{saving_throw.name}</li>
        ))}
      </ul>
    </div>
  );
};

export default SavingThrowsList;
