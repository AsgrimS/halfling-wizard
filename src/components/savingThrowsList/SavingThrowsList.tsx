import React from "react";
import { SavingThrowsData } from "../../graphql/interfaces";

const SavingThrowsList: React.FC<SavingThrowsData> = ({ saving_throws }) => {
  return (
    <div data-testid="saving-throws-list">
      <span>Saving Throws:</span>
      <ul>
        {saving_throws.map((saving_throw) => (
          <li key={saving_throw.index}>{saving_throw.name}</li>
        ))}
      </ul>
    </div>
  );
};

export default SavingThrowsList;
