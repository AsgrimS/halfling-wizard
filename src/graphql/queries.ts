import { gql } from "@apollo/client";

export const HERO_CLASS = gql`
  query GetHeroesClasses($classIndex: String!) {
    class(filter: { name: $classIndex }) {
      hit_die
      index
      name
      proficiencies {
        index
        name
      }
      saving_throws {
        index
        name
      }
      proficiency_choices {
        choose
        from {
          name
          index
        }
      }
      spellcasting {
        level
        spellcasting_ability {
          index
          name
        }
      }
      starting_equipment {
        equipment {
          name
          index
        }
        quantity
      }
      starting_equipment_options {
        choose
        from {
          equipment {
            name
            index
          }
          quantity
        }
      }
      subclasses {
        index
        name
      }
    }
  }
`;

export const HEROES_CLASSES = gql`
  query GetHeroesClasses {
    classes {
      name
      index
    }
  }
`;
