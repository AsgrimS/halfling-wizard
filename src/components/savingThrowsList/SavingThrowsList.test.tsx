import React from "react";
import { render, screen, cleanup } from "@testing-library/react";

import SavingThrowsList from "./SavingThrowsList";
import { AbilityScoreSkills } from "../../graphql/schema";

describe("SavingThrowsList component", () => {
  afterEach(() => {
    cleanup();
  });

  it("should render a list of saving throws", () => {
    const mockSavingThrowsListData: AbilityScoreSkills[] = [
      { name: "INT", index: "int" },
      { name: "DEX", index: "dex" },
      { name: "CHA", index: "cha" },
    ];

    render(<SavingThrowsList saving_throws={mockSavingThrowsListData} />);
    const savingThrowListElement = screen.getByTestId("saving-throws-list");
    const liElements = savingThrowListElement.getElementsByTagName("li");

    expect(savingThrowListElement).toBeInTheDocument();
    expect(liElements.length).toEqual(3);

    for (let i = 0; i < liElements.length; i += 1) {
      expect(liElements[i]).toContainHTML(
        `<li>${mockSavingThrowsListData[i].name}</li>`
      );
    }

    expect(savingThrowListElement).toMatchSnapshot();
  });
});
