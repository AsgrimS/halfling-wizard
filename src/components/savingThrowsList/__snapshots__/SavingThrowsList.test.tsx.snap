// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavingThrowsList component should render a list of saving throws 1`] = `
<div
  data-testid="saving-throws-list"
>
  <span>
    Saving Throws:
  </span>
  <ul>
    <li>
      INT
    </li>
    <li>
      DEX
    </li>
    <li>
      CHA
    </li>
  </ul>
</div>
`;
